assign-doi-datacite/src/main/java/no/unit/nva/doi/datacite/clients/DataCiteClient.java

```diff
@@ -159,22 +160,34 @@ public void deleteDraftDoi(URI customerId, Doi doi) throws ClientException {
         }
     }
 
-    private DataCiteMdsConnection prepareAuthenticatedDataCiteConnection(URI customerId) {
-        return mdsConnectionFactory.getAuthenticatedConnection(customerId);
+    private HttpResponse<String> sendDraftDoiRequest(DataCiteRestConnection connection, String prefix)
+        throws IOException, InterruptedException, CreateDoiException {
+        HttpResponse<String> response = connection.createDoi();
+        if (isUnsuccessfulResponse(response)) {
+            throw handleUnsuccessfulResponse(prefix, response);
+        }
+        return response;
+    }
+
+    private CreateDoiException handleUnsuccessfulResponse(String prefix, HttpResponse<String> response) {
+        logger.error(ERROR_CREATING_DOI_TEMPLATE, prefix, response.statusCode(), response.body());
+        return new CreateDoiException(prefix, response.statusCode(), response.body());
+    }
+
+    private DataCiteMdsConnection prepareAuthenticatedMdsDataCiteConnection(URI customerId) {
+        return dataCiteApiConnectionFactory.getAuthenticatedMdsConnection(customerId);
+    }
+
+    private DataCiteRestConnection prepareAuthenticatedDataCiteRestConnection(
+        URI customerId, DataCiteMdsClientSecretConfig secretConfig) {
+        return dataCiteApiConnectionFactory.getAuthenticatedRestConnection(customerId, secretConfig);
```

Comment:
The constructor for `DataCiteConnectionFactory` needs to accept the `secret config` in its constructor. You shouldn't need to know any reference about `secret config` when you are using the factory to obtain an authenticated connection for a customer.